File: app/csharp/AccountingSystem.Api/Dtos/ErrorResponse.cs
namespace AccountingSystem.Api.Dtos;

/// <summary>
/// エラーレスポンス DTO
/// </summary>
public class ErrorResponse
{
    /// <summary>
    /// HTTP ステータスコード
    /// </summary>
    public int Status { get; set; }

    /// <summary>
    /// エラー種別
    /// </summary>
    public string Error { get; set; } = string.Empty;

    /// <summary>
    /// エラーメッセージ
    /// </summary>
    public string Message { get; set; } = string.Empty;

    /// <summary>
    /// 詳細情報（バリデーションエラーなど）
    /// </summary>
    public List<string>? Details { get; set; }

    /// <summary>
    /// エラー発生日時
    /// </summary>
    public DateTime Timestamp { get; set; }
}

File: app/fsharp/AccountingSystem.Api/Dtos/ErrorResponse.fs
namespace AccountingSystem.Api.Dtos

open System

/// <summary>
/// エラーレスポンス DTO
/// </summary>
[<CLIMutable>]
type ErrorResponse = {
    /// HTTP ステータスコード
    Status: int
    /// エラー種別
    Error: string
    /// エラーメッセージ
    Message: string
    /// タイムスタンプ
    Timestamp: DateTime
}

module ErrorResponse =
    /// <summary>
    /// エラーレスポンスを作成
    /// </summary>
    let create status error message =
        {
            Status = status
            Error = error
            Message = message
            Timestamp = DateTime.UtcNow
        }
